CMakeLists.txt

```diff
@@ -34,8 +35,11 @@ check_cxx_compiler_flag(-fopenmp      CXX_HAVE_OMP)
 check_cxx_compiler_flag(-fopenmp-simd CXX_HAVE_OMP_SIMD)
 
 # Build options
-option(BUILD_TEST "Build googletest test suite" ON)
-option(BUILD_BENCHMARK "Build benchmark suite" ON)
+option(NODEPS_BUILD "Disable components that require extra external libraries to be present in the system. Effectively, it builds only the DALI core and kernel libraries")
+set(DEPS_ENABLED NOT NODEPS_BUILD)
+
+option(BUILD_TEST "Build googletest test suite" DEPS_ENABLED)  # Tests use OpenCV...
```

Comment:
I agree cmake_dependent_option should be here (I missed that). However, the issue with the test code is that several of them use OpenCV. I don't have the time to invest to disable just those tests that depend on OpenCV so that's why I am switching them off.

Reply:
Even if we can make it working I don't thinking we are able to maintain all combinations of test and options.

Reply:
Just saying, that this
```cmake
set(DEPS_ENABLED NOT NODEPS_BUILD)
option(BUILD_TEST "Build googletest test suite" DEPS_ENABLED)
```
looks strange and `cmake_dependent_option` can be used as well. I totally understand, why it's turned off ;)